### ISSUE TITLE
stat: Quantile returns wrong value when dataset is of even size

### ISSUE THREAD

**Issue by sbinet** [gonum/gonum#253]
according to the universal source of truth:
 https://en.wikipedia.org/wiki/Median

the median of:
```go
xs := []float64{1, 3, 3, 6, 7, 8, 9}
```
should be `6`.

this program:
```go
package main

import (
	"fmt"

	"gonum.org/v1/gonum/stat"
)

func main() {
	xs := []float64{1, 3, 3, 6, 7, 8, 9}
	med := stat.Quantile(0.5, stat.Empirical, xs, nil)
	fmt.Printf("median=%v\n", med)
}
```

indeed prints:
```
$> go run ./main.go
median=6
```

but, for this dataset: `xs := []float64{1, 2, 3, 4, 5, 6, 8, 9}`, it should return `(4 + 5) ÷ 2` _ie_: `4.5`.

`gonum/stat.Quantile` returns `4`:

```go
package main

import (
	"fmt"

	"gonum.org/v1/gonum/stat"
)

func main() {
	xs := []float64{1, 2, 3, 4, 5, 6, 8, 9}
	med := stat.Quantile(0.5, stat.Empirical, xs, nil)
	fmt.Printf("median=%v\n", med)
}
```

```
$> go run ./main.go 
median=4
```

**Comment by btracey** [comment]
The documentation could probably be improved, but this isn't a bug. This is the behavior set by CumulantKind. There are a bunch of different kinds https://en.wikipedia.org/wiki/Quantile#Estimating_quantiles_from_a_sample that can be implemented. You can also see our initial discussion here: https://github.com/gonum/stat/pull/1

**Comment by sbinet** [comment]
ok. so perhaps have a dedicated `stat.Median` function, then?

**Comment by btracey** [comment]
For example, I think the behavior you want is R2 on that list. Presumably in the weighted case it should do a weighted averaging.

**Comment by sbinet** [comment]
ok. `R2Cumulant` ? or just `R2` ?

**Comment by btracey** [comment]
There must be a better name than that. `AvgDiscont` (which is what it does)?

**Comment by sbinet** [comment]
SGTM

**Comment by sbinet** [comment]
see gonum/gonum#254.

**Comment by btracey** [comment]
This was solved by https://github.com/gonum/gonum/pull/605 , right?

**Comment by sbinet** [comment]
Fixed by #605, yes.
